src/main/java/ch/njol/skript/entity/EntityData.java

```diff
@@ -503,10 +522,13 @@ public E spawn(Location location, org.bukkit.util.@Nullable Consumer<E> consumer
 	@Nullable
 	public E spawn(Location location, @Nullable Consumer<E> consumer) {
 		assert location != null;
+		World world = location.getWorld();
+		if (world == null || !canSpawn(world))
```

Comment:
Can we check for null world inside canSpawn method instead? that should be better for the future :)

Reply:
sounds good to me